### PR TITLE
media-started-playing & media-paused refactor

### LINKED ISSUE #12382: continually prompted to play YT videos
### Description
If you have 'Autoplay Media' set to 'Always ask' and you do not select 'remember this decision', you will be continually prompted to 'Allow youtube to autoplay media?' and can't actually view the video.

### Steps to Reproduce

MacOS steps:
  1. Clean profile on 0.20.9
  2. Navigate to about:preferences#security
  3. Set Autoplay Media to 'Always ask'
  4. Clear any pre-set settings for Autoplay Media (for YT and Twitch)
  5. Navigate to YT, select a video to display
  6. Notification displays, select Allow.
  7. Video plays for a split second and then you get the notification again.
  8. Navigate to a different site to view a video (I used an nytimes.com article with a video)
  9. Notification displays, select Allow.
  10. Video plays without issue.

Win/Linux steps (slightly different):
  1. Clean profile on 0.20.9
  2. Navigate to about:preferences#security
  3. Set Autoplay Media to 'Always ask'
  4. Clear any pre-set settings for Autoplay Media (for YT and Twitch)
  5. Navigate to YT, select a video to display
  6. Video plays without any notification bar (this is pre-existing on 0.19.122)
  7. Restart browser, navigate to YT video.
  6. Notification displays, select Allow.
  7. Video plays for a split second and then you get the notification again.
  8. Navigate to a different site to view a video (I used an nytimes.com article with a video)
  9. Notification displays, select Allow.
  10. Video plays without issue.

**Actual result:**
Unless you select 'Remember this decision' you will be continually prompted for YT videos and cannot actually view them. 

play-notif

**Expected result:**
You should only have to select 'Allow' once and then the YT video should play. (this is how this works on 0.19.122

**Reproduces how often:**
Easily

### Brave Version

**about:brave info:**
0.20.9

**Reproducible on current live release:**
no, not reproducible on 0.19.122

### Additional Information
See slightly different Win/Linux steps which were confirmed by @srirambv and @kjozwiak

### PR DESCRIPTION
**Pull request by darkdh**
fix #12382

Auditors: @bridiver, @bsclifton

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
a. Test speaking icon
a.1 Play any video with sounds
a.2 There should be speaking indicator on tab
a.3 Pause the video
a.4 The icon in a.2 should be gone

b Temporary allow for autoplay
b.0 Make sure there is no autoplay rule for test site (youtube)
b.1 Play any videoes
b.2 Autoplay banner will show up
b.3 Click allow and not remember
b.4 Video can continue playing
b.5 Check about:preferences#security, there should be autoplay rule for it
b.6 Close the tab
b.7 Check about:preferences#security, the rule should be gone
b.8 Repeat b.1 - b.5
b.9 Relaunch Brave
b.10 Check about:preferences#security, the rule should be gone

c Temporary allow for autoplay (ignore banner)
c.0 Make sure there is no autoplay rule for test site (youtube)
c.1 Play any videoes
c.2 Autoplay banner will show up
c.3 Click play button on video
c.4 Video can continue playing (banner will be dismissed)
c.5 Check about:preferences#security, there should be autoplay rule for it
c.6 Close the tab
c.7 Check about:preferences#security, the rule should be gone
c.8 Repeat c.1 - c.5
c.9 Relaunch Brave
c.10 Check about:preferences#security, the rule should be gone

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
Added @cezaraugusto as a reviewer after not seeing the audio icon show up in the tab

**Comment by bsclifton** [comment]
Verified the audio icon works as expected in 0.20.x (ex: see 0.20.12 packaged build).  However, when this commit is cherry-picked into 0.20.x, it does not show the audio icon when a video is played via allowing an auto-played video

**Comment by darkdh** [comment]
fixed in c0b17ae98b3ad6ba06dcc2dda7890593754b638c

File: app/renderer/reducers/frameReducer.js [hunk 1]
```diff
@@ -190,6 +190,19 @@ const frameReducer = (state, action, immutableAction) => {
         state = frameStateUtil.updateTabPageIndex(state, tabId)
       }
       break
+
+    case appConstants.APP_MEDIA_STARTED_PLAYING: {
+      const index = frameStateUtil.getIndexByTabId(state, action.tabId)
+      state = state.setIn(['frames', index, 'audioPlaybackActive'], true)
```
Comment (bsclifton) [inline app/renderer/reducers/frameReducer.js]: thinking

**Comment by bsclifton** [comment]
master 172d6599642a9d4d7d51c3fb367924bf36f3ad0f
0.21.x 82bbd867c76fa56b2e8e8cdda3dc4a9e26ed5be0
0.20.x ff764a62524e4d886babd27d64e2fc6fffeb84d6

**Pull request merged**